nixos/tests/nextcloud/basic.nix

```diff
@@ -13,10 +13,17 @@ in {
     # The only thing the client needs to do is download a file.
     client = { ... }: {
       services.davfs2.enable = true;
-      system.activationScripts.davfs2-secrets = ''
-        echo "http://nextcloud/remote.php/dav/files/${adminuser} ${adminuser} ${adminpass}" > /tmp/davfs2-secrets
-        chmod 600 /tmp/davfs2-secrets
-      '';
+      systemd.services.davfs2-secrets = {
+        wantedBy = [ "sysinit.target" ];
+        before = [ "sysinit.target" ];
+        unitConfig.DefaultDependencies = false;
+        serviceConfig.Type = "oneshot";
+        serviceConfig.RemainAfterExit = true;
+        script = ''
+          echo "http://nextcloud/remote.php/dav/files/${adminuser} ${adminuser} ${adminpass}" > /tmp/davfs2-secrets
+          chmod 600 /tmp/davfs2-secrets
```

Comment:
out of scope for this PR but unless the service is using private tmp then this is not that secure